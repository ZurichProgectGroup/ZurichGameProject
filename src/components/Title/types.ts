export type OwnProps = {
    className?: string,
    text: string,
    tagName?: keyof JSX.IntrinsicElements
};

export type Props = OwnProps;
