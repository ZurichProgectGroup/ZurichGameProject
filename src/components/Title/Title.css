.title {
    font-weight: 600;
    font-size: 36px;
    line-height: 43px;
    text-align: center;
    color: var(--font-color);
    text-shadow: 3px 2px 0 var(--main);
    text-transform: uppercase;
}
