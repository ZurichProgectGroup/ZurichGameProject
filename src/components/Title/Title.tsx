import React from 'react';
import cn from 'classnames';
import type { Props } from './types';
import './Title.css';

const Title = (
    {
        text,
        className = '',
        tagName: TagName = 'b',
    } : Props,
) => (
    <TagName className={cn('title', className)}>{text}</TagName>
);

export default Title;
