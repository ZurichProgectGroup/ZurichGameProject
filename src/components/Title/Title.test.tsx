import React from 'react';
import { render } from '@testing-library/react';
import Title from './Title';

it('renders correctly', () => {
    const { asFragment } = render(<Title text="test title" />);

    expect(asFragment()).toMatchSnapshot();
});
