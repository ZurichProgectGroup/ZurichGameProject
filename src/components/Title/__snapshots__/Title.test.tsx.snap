// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <b
    class="title"
  >
    test title
  </b>
</DocumentFragment>
`;
